packages/browser/src/eventbuilder.ts

```diff
@@ -283,3 +283,37 @@ export function eventFromString(
 
   return event;
 }
+
+function getNonErrorObjectExceptionValue(
+  exception: Record<string, unknown>,
+  { isUnhandledRejection }: { isUnhandledRejection?: boolean },
+): string {
+  const keys = extractExceptionKeysForMessage(exception);
+  const captureType = isUnhandledRejection ? 'promise rejection' : 'exception';
+  const className = getObjectClassName(exception);
+
+  // Some ErrorEvent instances do not have an `error` property, which is why they are not handled before
+  // We still want to try to get a decent message for these cases
+  if (isErrorEvent(exception)) {
+    return `Event \`ErrorEvent\` captured as ${captureType} with message \`${exception.message}\``;
+  }
+
+  const name = isEvent(exception)
+    ? `Event \`${className}\` (${exception.type})`
+    : className && className !== 'Object'
+    ? `\`${className}\``
+    : 'Object';
+
+  const label = `${name} captured as ${captureType}`;
+
+  return `${label} with keys: ${keys}`;
+}
+
+function getObjectClassName(obj: unknown): string | undefined | void {
+  try {
+    const prototype: Prototype | null = Object.getPrototypeOf(obj);
+    return prototype ? prototype.constructor.name : undefined;
```

Comment:
I am worried about this for similar reasons outlined in: https://github.com/getsentry/sentry-javascript/pull/8161#issuecomment-1567937052

If people throw an object of a custom class and their code is minified, `prototype.constructor.name` will contain the minified name. Minified symbols may change across deploys and grouping may therefore break across deploys.
The current implementation doesn't have this issue as object keys are generally not minified.

I don't know yet how I would proceeed here though 🤔

Reply:
Hmm, this is true (tried it in a test app). What about the following - slightly hacky but maybe OK - solution:

```js
if (className.length === 1) {
  // this is most likely minified/mangled, so just use "Object"
  return "Object"
}
```

? This will not capture _everything_ (you may have mangled class names with more than one char), but usually mangling will reduce to one character as it is the smallest, and you rarely have more than 26 entities in the same scope 🤔 Or possibly we could even do `className.length < 3` or so, as it is not necessarily a problem if we don't capture one or the other "real" class name, I guess.

Reply:
I wouldn't do that. That seems a bit silly.

The more I think about this, the more I wouldn't change anything. We're just introducing new complexity that doesn't solve any particular issue.

The one thing I would consider is either updating the Error message to be a bit more clear what is happening or removing the "with keys ..." part and using `scope.setFingerprint()` to keep the grouping intact.

Reply:
I would shy away from using the fingerprint because it make it harder for users to understand how to filter the events out and understand how they are being grouped.

How about let's just not differentiate between class instances and plain js object?

Reply:
OK, for simplicity I will just go with `Object` for everything except events then? That should already be better than what we have right now and we can still tweak/further improve this later. (FWIW, I still think it would be helpful to get the class name, but we can think this over some more)

Reply:
yup sounds good to me - I think we can keep iterating